meinberlin/apps/kiezradar/serializers.py

```diff
@@ -0,0 +1,111 @@
+from rest_framework import serializers
+
+from adhocracy4.administrative_districts.models import AdministrativeDistrict
+from adhocracy4.projects.models import Topic
+from meinberlin.apps.kiezradar.models import KiezradarQuery
+from meinberlin.apps.kiezradar.models import ProjectType
+from meinberlin.apps.kiezradar.models import SearchProfile
+from meinberlin.apps.organisations.models import Organisation
+
+
+class SearchProfileSerializer(serializers.ModelSerializer):
+    """Serializer for the SearchProfile model."""
+
+    query = (
+        serializers.PrimaryKeyRelatedField(
+            queryset=KiezradarQuery.objects.all(), required=False
+        ),
+    )
+    query_text = serializers.CharField(required=False)
+    organisations = serializers.PrimaryKeyRelatedField(
+        queryset=Organisation.objects.all(), many=True, allow_null=True, required=False
+    )
+    districts = serializers.PrimaryKeyRelatedField(
+        queryset=AdministrativeDistrict.objects.all(),
+        many=True,
+        allow_null=True,
+        required=False,
+    )
+    project_types = serializers.PrimaryKeyRelatedField(
+        queryset=ProjectType.objects.all(), many=True, allow_null=True, required=False
+    )
+    topics = serializers.PrimaryKeyRelatedField(
+        queryset=Topic.objects.all(), many=True, allow_null=True, required=False
+    )
+    status_display = serializers.CharField(source="get_status_display", read_only=True)
+
+    class Meta:
+        model = SearchProfile
+        fields = [
+            "id",
+            "user",
+            "name",
+            "description",
+            "disabled",
+            "status",
+            "status_display",
+            "query",
+            "query_text",
+            "organisations",
+            "districts",
+            "project_types",
+            "topics",
+        ]
+
+        read_only_fields = ["user"]
+
+    def create(self, validated_data):
+        # Pop many-to-many and one-to-many fields from validated_data
+        query_text = validated_data.pop("query_text", None)
+        organisations_data = validated_data.pop("organisations", None)
+        districts_data = validated_data.pop("districts", None)
+        project_types_data = validated_data.pop("project_types", None)
+        topics_data = validated_data.pop("topics", None)
+
+        if query_text:
+            query, _ = KiezradarQuery.objects.get_or_create(text=query_text)
+            validated_data["query"] = query
+
+        # need to create a profile before we can add m2m relations
+        search_profile = SearchProfile.objects.create(**validated_data)
```

Comment:
Instead of popping and then setting the m2m relations manually below, can we just call `return super().create(validated_data)` here as it will set m2m relations for us ?

Reply:
the search profile has to be created first in order to add m2m relations

Reply:
yes, but the super call does that as far as I can see: https://github.com/encode/django-rest-framework/blob/dbac145638758413b966c3418fa5f3f651e3e02a/rest_framework/serializers.py#L989

Reply:
Will try again.

Reply:
if it doesn't work we can also proceed like it is!

Reply:
I did, last time I tried I forgot to remove the line where I also save it manually